src/ParallelAlgorithms/Amr/Projectors/Tensors.hpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/ApplyMatrices.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Projection.hpp"
+#include "ParallelAlgorithms/Amr/Protocols/Projector.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace amr::projectors {
+
+/// \brief Update the Tensors corresponding to TensorTags after an AMR
+/// change
+template <size_t Dim, typename... TensorTags>
+struct ProjectTensors : tt::ConformsTo<amr::protocols::Projector> {
+  using return_tags = tmpl::list<TensorTags...>;
```

Comment:
Question: Does the check that all tags are projected work with Variables and subitems?

Reply:
The DataBox is responsible for handling the subitems.   AMR projectors cannot be called on subitems.